examples/seismic/tutorials/04_dask.ipynb

```diff
@@ -0,0 +1,269 @@
+    "from examples.seismic import demo_model\n",
+    "from examples.seismic import RickerSource, Receiver\n",
+    "\n",
+    "import numpy as np\n",
+    "\n",
+    "# Define true and initial model\n",
+    "def get_grid():\n",
+    "    shape = (101, 101)  # Number of grid point (nx, nz)\n",
+    "    spacing = (10., 10.)  # Grid spacing in m. The domain size is now 1km by 1km\n",
+    "    origin = (0., 0.)  # Need origin to define relative source and receiver locations\n",
+    "\n",
+    "    return shape, spacing, origin\n",
+    "\n",
+    "def get_true_model():\n",
+    "    shape, spacing, origin = get_grid()\n",
+    "    return demo_model('circle-isotropic', vp=3.0, vp_background=2.5, \n",
+    "                      origin=origin, shape=shape, spacing=spacing, nbpml=40)\n",
+    "\n",
+    "def get_initial_model():\n",
+    "    shape, spacing, origin = get_grid()\n",
+    "\n",
+    "    return demo_model('circle-isotropic', vp=2.5, vp_background=2.5, \n",
+    "                      origin=origin, shape=shape, spacing=spacing, nbpml=40)\n",
+    "\n",
+    "def dump_model(param, model):\n",
+    "    np.save(param['model'], model.astype(np.float32))\n",
+    "\n",
+    "def load_model(param):\n",
+    "    return np.load(param['model'])\n",
+    "\n",
+    "def get_current_model(param):\n",
+    "    model = get_initial_model()\n",
+    "    model.m.data[:] = np.reshape(load_model(param), model.m.data.shape)\n",
+    "    return model\n",
+    "\n",
+    "# Define acquisition geometry: source\n",
+    "def get_data(param):\n",
+    "    \"\"\" Returns source and receiver data for shot i.\n",
+    "    \"\"\"\n",
+    "    true_model = get_true_model()\n",
+    "    dt = true_model.critical_dt  # Time step from model grid spacing\n",
+    "\n",
+    "    # Set up src\n",
+    "    nt = int(1 + (param['tn']-param['t0']) / dt)  # Discrete time axis length\n",
+    "\n",
+    "    src = RickerSource(name='src', ndim=2, f0=param['f0'],\n",
+    "                       time=np.linspace(param['t0'], param['tn'], nt))\n",
+    "    src.coordinates.data[0, :] = [30, param['shot_id']*1000./(param['nshots']-1)]\n",
+    "\n",
+    "    # Set up rec\n",
+    "    nreceivers = 101  # Number of receiver locations per shot\n",
+    "    rec = Receiver(name='rec', npoint=nreceivers, ntime=nt, ndim=2)\n",
+    "    rec.coordinates.data[:, 1] = np.linspace(0, true_model.domain_size[0], num=nreceivers)\n",
+    "    rec.coordinates.data[:, 0] = 980. # 20m from the right end\n",
+    "\n",
+    "    # Set up solver - using model_in so that we have the same dt\n",
+    "    solver = AcousticWaveSolver(true_model, src, rec, space_order=4)\n",
+    "\n",
+    "    # Generate synthetic receiver data from true model\n",
+    "    true_d, _, _ = solver.forward(src=src, m=true_model.m)\n",
+    "\n",
+    "    return src, true_d, nt, solver\n",
+    "\n",
+    "# Create FWI gradient kernel \n",
+    "def fwi_gradient_i(work):\n",
+    "    clear_cache()\n",
```

Comment:
comment why this is necessary

Reply:
creates new `u0` each call but is not cleared automatically for some reason, so memory overload without it.